### ISSUE TITLE
Plasma 6: Pinned kde application icons go blank after a while

### ISSUE THREAD

**Issue by chewblacka** [NixOS/nixpkgs#308252]
### Describe the bug
When pinning icons to the Plasma 6 Task Manager (i.e. task bar), after a while (days, weeks) the icons get corrupted. E.g. a shortcut to `konsole` becomes renamed to `org.kde.konsole.desktop` and the icon becomes blank. Both left-click and right-click still work as expected. Left-click will open the respective program and whilst running, the blank icon will be replaced by the correct icon. When the program is closed it returns to a blank icon.

Issue can be mitigated by unpinning then re-pinning the icons, but as this is a regular occurrence this is somewhat annoying.

### Steps To Reproduce
1. In Plasma 6 pin a few short-cuts to the Task Manager
2. Wait a few days to a week or two

### Expected behavior
For the icons to not get corrupted.

### Screenshots
![plasma-blank-icons](https://github.com/NixOS/nixpkgs/assets/18430320/b6f2c3d0-d9e7-40d9-a85c-e22b6def5ca8)

### Notify maintainers
@K900 @NixOS/qt-kde 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.27, NixOS, 24.05 (Uakari), 24.05.20240416.5672bc9`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - channels(root): `""`
 - channels(ch): `""`
 - nixpkgs: `/nix/store/3kwj19dbdfxnjbcns4hw307ylhz3wgrm-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by chewblacka** [comment]
Ok many thanks! I'll mark this as closed.

**Comment by sakuexe** [comment]
If someone comes across this on Google, I found a work around for this. The original fix was from here: [Plasma 6.1.3: Pinned kde application icons go blank after GC(NixOS)](https://discuss.kde.org/t/plasma-6-1-3-pinned-kde-application-icons-go-blank-after-gc-nixos/19444)

This command will replace the `/nix/store/.../applications/` parts of the icons with `applications:`. This way they are not tied to a specific generation of the packages. It will also reload the plasmashell, so that the changes should be visible right away.

```bash
sed -i 's/file:\/\/\/nix\/store\/[^\/]*\/share\/applications\//applications:/gi' \
      ~/.config/plasma-org.kde.plasma.desktop-appletsrc \
      && systemctl restart --user plasma-plasmashell
```
